tests/mock_network/churn.rs

```diff
@@ -571,30 +577,34 @@ fn messages_during_churn() {
             max_iterations,
             current_sections(&nodes).format(", ")
         );
-        let new_indices = random_churn(
+        let (added_indices, dropped_names) = random_churn(
             &mut rng,
             &network,
             &mut nodes,
             churn_probability,
             min_section_num,
             max_section_num,
         );
+
+        if !dropped_names.is_empty() {
+            warn!("Dropping nodes: {:?}", dropped_names);
+        }
+
         let relocation_map = RelocationMapBuilder::new(&nodes);
         let expectations = setup_expectations(&mut rng, &mut nodes, elder_size);
 
         poll_and_resend(&mut nodes);
 
         let relocation_map = relocation_map.build(&nodes);
 
-        let added_names = check_added_indices(&mut nodes, new_indices);
+        let added_names = check_added_indices(&mut nodes, added_indices);
         if !added_names.is_empty() {
             warn!("Added nodes: {:?}", added_names);
         }
 
-        shuffle_nodes(&mut rng, &mut nodes);
-
         expectations.verify(&mut nodes, &relocation_map);
         verify_invariant_for_all_nodes(&network, &mut nodes);
+        shuffle_nodes(&mut rng, &mut nodes);
```

Comment:
Why did the shuffle move from before verify to after?

Reply:
No strong reason, I just wanted to have the verification bits together. Don't think it changes anything.